Sming/Services/Yeelight/YeelightBulb.cpp

```diff
@@ -157,32 +156,31 @@ bool YeelightBulb::onResponse(TcpClient& client, char* data, int size)
 			p2 = source.length();
 		String buf = source.substring(p, p2);
 		p = unsigned(p2) + 2;
-		DynamicJsonBuffer jsonBuffer;
-		JsonObject& root = jsonBuffer.parseObject(buf);
-		bool parsed = root.success();
-		if (parsed)
+		DynamicJsonDocument doc(1024);
+		auto error = deserializeJson(doc, buf);
+		if (!error)
 		{
-			if (root.containsKey("id") && root.containsKey("result"))
+			if (doc.containsKey("id") && doc.containsKey("result"))
 			{
-				long id = root["id"];
+				long id = doc["id"];
 				if (id == propsId)
 				{
-					auto &result = root["result"].asArray();
-					String resp = result[0].asString();
+					JsonArray result = doc["result"];
+					String resp = result[0];
 					parsePower(resp);
 				}
 			}
-			if (root.containsKey("method") && root.containsKey("params"))
+			if (doc.containsKey("method") && doc.containsKey("params"))
 			{
-				String method = root["method"].asString();
+				String method = doc["method"];
```

Comment:
should be `doc["method"].as<const char*>()`

Reply:
As far as I understood from ["Automatic conversion"](https://arduinojson.org/v6/doc/upgrade/) that should happen automatically and there is not extra need to cast it to string.

Reply:
https://github.com/bblanchon/ArduinoJson/issues/420

Reply:
```
void testAsString()
{
	StaticJsonDocument<256> doc;
	doc["exists"] = "abc";
	String s1 = doc["method"];
	debug_i("S1 = %d chars '%s'", s1.length(), s1.c_str());
	String s2 = doc["method"].as<const char*>();
	debug_i("S2 = %d chars '%s'", s2.length(), s2.c_str());
	String s3 = (const char*)doc["method"];
	debug_i("S3 = %d chars '%s'", s3.length(), s3.c_str());
}
```
Output:
```
4642436 S1 = 4 chars 'null'
4642528 S2 = 0 chars ''
4642608 S3 = 0 chars ''
```

Reply:
The most portable and explicit way is to use [JsonVariant::operator|](https://arduinojson.org/v6/api/jsonvariant/or/) to specify the default value:

```c++
String value = doc["key"] | "";
```

See a comparative demo here:
https://wandbox.org/permlink/oxMsD1f3qYFq9jC0

Reply:
That's a much cleaner solution. Thank you.